docs/user-guide.md

```diff
@@ -15,37 +15,35 @@ Such annotations are useful to (for example):
 
 ## Java variables are references
 
-In Java, all variables, except primitives like `int`, are references. We often
-think of `String x;` as meaning "`x` is a `String`", but actually it means "`x` is a
-_reference_ to a `String`". One important difference between these two views is that
-`x` can also be null. The goal of JSpecify is to move towards a world where `String
-x;` means "`x` is a reference to an actual string", and if you want a reference
-that might be null then you write `@Nullable String x;`.
+In Java, all non-primitive variables are references. We often think of `String
+x;` as meaning "`x` is a `String`", but actually it means "`x` is a _reference_
+to a string or it is null". JSpecify includes a notion of "null-marked code"
```

Comment:
This might be slightly clearer by using "either": "either x is a reference to a string or it is null".

Reply:
I feel like this one sentence has generated more discussion than everything else. :-) I've piled on yet let more words in an attempt to make the point clear, because I do think it is an important point that sets up the rest of the document.